test/error-messages/nested-invalid-value.txt

```diff
@@ -0,0 +1,9 @@
+Invalid value
+
+concat :: Array a -> Array a -> Array a
+                     ^^^^^^^
+                        1
+
+1)  "ab" :: String
+
+The value at position 1 is not a member of (Array Number).
```

Comment:
I was planning to look into this today. Do you see a very simple way to know when we should do this? My first impression was that this would not be trivial..